### PR TITLE
Add publish-godoc Make target

### PR DESCRIPTION
**Pull request by pellared**
## Why

Automate the release

## What

Add `publish-godoc` Make target to facilitate updating the https://pkg.go.dev/github.com/signalfx/signalfx-go-tracing content.

## Testing

Was used for `v1.10.0`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,6 +54,17 @@ push-tag: # example usage: make push-tag tag=v1.100.1 remote=origin
 	 	git push $(remote) "$${dir:2}/$(tag)"); \
 	done
 
+.PHONY: publish-godoc
+publish-godoc: # example usage: make publish-godoc tag=v1.100.1
+	@[ "$(tag)" ] || ( echo ">> 'tag' is not set"; exit 1 )
+	@echo "godoc for root"
+	@cd ; go get -u github.com/signalfx/signalfx-go-tracing@$(tag)
+	@set -e; cd; \
```
Comment (MrAlias) [inline Makefile:62]: What are the `cd` commands doing here?

File: Makefile [hunk 2]
```diff
@@ -54,6 +54,17 @@ push-tag: # example usage: make push-tag tag=v1.100.1 remote=origin
...
+	for dir in $(SUBMODULES); do \
+	  (echo godoc for "$${dir:2}" && \
```
Comment (MrAlias) [inline Makefile:64]:
```suggestion
	  (echo "godoc for $${dir:2}" && \
```

File: Makefile (hunk 1 above)
Comment (pellared) [reply Makefile:62]: Changing the directory to home so that the `go.mod` is NOT modified

Comment (MrAlias) [reply Makefile:62]: Do we need to do this twice? Also, to ensure things in the home directory are not modified, should we change to a newly created temp directory?

Comment (pellared) [reply Makefile:62]: No. We can do it once 👍 

Never seen anyone to have `go.mod` in `$HOME`. The `cd && go get` pattern is pretty common. See: https://maelvls.dev/go111module-everywhere/ . As far as I remember [VS Code Go extension](https://github.com/golang/vscode-go) is using it to install the Go tools.

File: Makefile (hunk 2 above)
Comment (pellared) [reply Makefile:64]: I will have to test it. I see the same issue in other Make targets.

Comment (pellared) [reply Makefile]: Done https://github.com/signalfx/signalfx-go-tracing/pull/159/commits/7c058849238db83bcc0348dc5ab6dede7e549e84

File: Makefile (hunk 1 above)
Comment (MrAlias) [reply Makefile:62]: > The cd && go get pattern is pretty common.

Understood. Doing this in a temp directory will avoid the issue entirely and is a pattern for us to start setting as a good example for others. :smile: 

You should be able to support both Linux and BSD variants with:

```sh
tmpdir=$(mktemp -d 2>/dev/null || mktemp -d -t 'goget')
```

Comment (pellared) [reply Makefile]: PTAL https://github.com/signalfx/signalfx-go-tracing/pull/159/commits/50582022f7ba163d2d9a47d147d2ea12b26c2f75

**Review by MrAlias — APPROVED** [review]

**Pull request merged**